app/src/main/java/com/example/bottomnav/VariableContents.java

```diff
@@ -1,20 +1,121 @@
+     * @return Optional<Components>
+     * @throws Exception
+     *
+     * Difference: Simple specifiers don't quantifier types and do not have name object
+     */
+    private static Optional<VariableContents> makeOptFromDecl(CPPASTSimpleDeclSpecifier simpleSpec,
+                                                              CPPASTDeclarator declarator) throws Exception {
+        CPPASTEqualsInitializer init = (CPPASTEqualsInitializer) declarator.getInitializer();
+        Optional<CPPASTEqualsInitializer> initOpt = makeOptInit(init);
+
+        if (declarator != null && simpleSpec != null) {
+            String declaratorNameStr = declarator.getName().getRawSignature();
+            String typeQualiferNameStr = simpleSpec.getSyntax().getImage();
+            String primitiveNameStr = getPrimitiveType(simpleSpec.getType());
+            return Optional.of(new VariableContents(declaratorNameStr, typeQualiferNameStr,
+                    primitiveNameStr, initOpt));
+        } else {
+            return Optional.empty();
+        }
+    }
+
+    /**
+     *
+     * @param namedTypeSpec
+     * @param declarator
+     * @return Optional<Components>
+     * @throws Exception
+     *
+     * Difference: Name typed specifiers have quantifer types and have name object
+     */
+    private static Optional<VariableContents> makeOptFromDecl(CPPASTNamedTypeSpecifier namedTypeSpec,
+                                                              CPPASTDeclarator declarator) throws Exception {
+        CPPASTEqualsInitializer init = (CPPASTEqualsInitializer) declarator.getInitializer();
+        Optional<CPPASTEqualsInitializer> initOpt = makeOptInit(init);
+        if (declarator != null && namedTypeSpec != null) {
+            String declaratorNameStr = declarator.getName().getRawSignature();
+            String typeQualifierNameStr = namedTypeSpec.getSyntax().getImage();
+            String primitiveTypeNameStr = namedTypeSpec.getName().getRawSignature();
+            return Optional.of(new VariableContents(declaratorNameStr, typeQualifierNameStr,
+                    primitiveTypeNameStr, initOpt));
+        } else {
+            return Optional.empty();
+        }
+    }
+
+    private static Optional<CPPASTEqualsInitializer> makeOptInit(CPPASTEqualsInitializer init) {
+        if (init != null) {
+            return Optional.of(init);
+        } else {
+            return Optional.empty();
+        }
+    }
+
+    private Optional<String> getValue(Optional<CPPASTEqualsInitializer> givenInit) {
+        if (givenInit.isPresent()) {
+            return Optional.of(((CPPASTLiteralExpression) givenInit.get().getInitializerClause()).getRawSignature());
+        }
+        return Optional.empty();
     }
 
-    public void setPacketSize(int size) {
-        packetSize = size;
+    /**
+     * @param primType
+     * @return char[]
+     *
+     * AST parser retrieves an integer to represent a primitive type. This function allows to
```

Comment:
Is there a function in the parser / AST library that does this? I'd expect (more likely than not, but not 100%) that there would be some utility to turn those numbers into something readable.

If you must do this yourself, consider having an enum so it's more type safe.